Source/FFmpegMediaSource.cpp

```diff
@@ -741,11 +741,27 @@ namespace winrt::FFmpegInteropX::implementation
             }
 
             //inject custom properties
-            if (config->AutoCorrectAnsiSubtitles() && config->IsExternalSubtitleParser && streamByteOrderMark != ByteOrderMark::UTF8)
+            if (config->AutoCorrectAnsiSubtitles() && config->IsExternalSubtitleParser &&
+                (streamEncoding != TextEncodingDetect::UTF8_BOM && streamEncoding != TextEncodingDetect::UTF8_NOBOM && streamEncoding != TextEncodingDetect::None))
             {
                 hstring key = config->AnsiSubtitleEncoding().Name();
                 std::string keyA = StringUtils::PlatformStringToUtf8String(key);
-                const char* keyChar = keyA.c_str();
+                const char* keyChar;
+
+                switch (streamEncoding)
+                {
+                case TextEncodingDetect::UTF16_LE_BOM:
+                case TextEncodingDetect::UTF16_LE_NOBOM:
+                    keyChar = "CP1200";
+                    break;
+                case TextEncodingDetect::UTF16_BE_BOM:
+                case TextEncodingDetect::UTF16_BE_NOBOM:
+                    keyChar = "CP1201";
```

Comment:
I'm not sure whether this will work, as ffmpeg uses libiconv for conversion and it doesn't state that it would support it?

https://www.gnu.org/software/libiconv/

Reply:
These were listed on a website related to ffmpeg and iconv. But maybe they had wrong info. We might have to use UTF-16BE, UTF-16LE instead, which are officially identifiers according to iconv website. I will try to do some testing with subs stored as UTF16.

Reply:
Do such subtitle files even exist? Maybe it isn't worth the effort..

Reply:
Conversion from UTF16 does not work, no matter if I use CP1200/CP12001 or UTF16-LE/UTF16-BE (the latter are official iconv encodings, but they don't work). So I dropped UTF16. I never saw it used in subs anyways.

Reply:
It's fine, not worth bothering with it.